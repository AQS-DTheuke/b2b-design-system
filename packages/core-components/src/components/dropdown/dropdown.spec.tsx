import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { DropdownComponent } from './dropdown';

it('should render the dropdown', async () => {
  const page = await newSpecPage({
    components: [DropdownComponent],
    template: () => <b2b-dropdown label="test"></b2b-dropdown>,
  });
  expect(page.root).toMatchSnapshot();
});

it('should render the dropdown with options', async () => {
  const page = await newSpecPage({
    components: [DropdownComponent],
    template: () => (
      <b2b-dropdown label="test">
        <option>An option</option>
      </b2b-dropdown>
    ),
  });
  expect(page.root).toMatchSnapshot();
});

it('should render a hint if a hint string is specified', async () => {
  const page = await newSpecPage({
    components: [DropdownComponent],
    template: () => (
      <b2b-dropdown label="test" hint="This is a test hint"></b2b-dropdown>
    ),
  });
  expect(page.root).toMatchSnapshot();
});

it('should disable the dropdown', async () => {
  const page = await newSpecPage({
    components: [DropdownComponent],
    template: () => <b2b-dropdown label="test" disabled></b2b-dropdown>,
  });
  expect(page.root).toMatchSnapshot();
});

it('should apply an error style to the host element if an error is specified and the select is set to invalid', async () => {
  const page = await newSpecPage({
    components: [DropdownComponent],
    template: () => (
      <b2b-dropdown
        label="test"
        error="This is a test error"
        invalid></b2b-dropdown>
    ),
  });
  expect(page.root).toMatchSnapshot();
});

it('should render an error message if an error is specified and the select is set to invalid', async () => {
  const page = await newSpecPage({
    components: [DropdownComponent],
    template: () => (
      <b2b-dropdown
        label="test"
        error="This is a test error"
        invalid></b2b-dropdown>
    ),
  });
  expect(page.root).toMatchSnapshot();
});

it('should render a hint if  invalid and disabled are true at the same time', async () => {
  const page = await newSpecPage({
    components: [DropdownComponent],
    template: () => (
      <b2b-dropdown
        label="test"
        error="This is a test error"
        hint="this is a test hint"
        invalid
        disabled></b2b-dropdown>
    ),
  });
  expect(page.root).toMatchSnapshot();
});
