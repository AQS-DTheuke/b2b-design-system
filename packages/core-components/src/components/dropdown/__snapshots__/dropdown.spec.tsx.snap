// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should apply an error style to the host element if an error is specified and the select is set to invalid 1`] = `
<b2b-dropdown class="b2b-dropdown b2b-dropdown--error" invalid="">
  <b2b-input-label>
    test
  </b2b-input-label>
  <select class="b2b-dropdown__select"></select>
  <span>
    This is a test error
  </span>
</b2b-dropdown>
`;

exports[`should disable the dropdown 1`] = `
<b2b-dropdown class="b2b-dropdown b2b-dropdown--disabled" disabled="">
  <b2b-input-label>
    test
  </b2b-input-label>
  <select class="b2b-dropdown__select" disabled=""></select>
</b2b-dropdown>
`;

exports[`should render a hint if  invalid and disabled are true at the same time 1`] = `
<b2b-dropdown class="b2b-dropdown b2b-dropdown--disabled" disabled="" invalid="">
  <b2b-input-label>
    test
  </b2b-input-label>
  <select class="b2b-dropdown__select" disabled=""></select>
  <span>
    this is a test hint
  </span>
</b2b-dropdown>
`;

exports[`should render a hint if a hint string is specified 1`] = `
<b2b-dropdown class="b2b-dropdown">
  <b2b-input-label>
    test
  </b2b-input-label>
  <select class="b2b-dropdown__select"></select>
  <span>
    This is a test hint
  </span>
</b2b-dropdown>
`;

exports[`should render an error message if an error is specified and the select is set to invalid 1`] = `
<b2b-dropdown class="b2b-dropdown b2b-dropdown--error" invalid="">
  <b2b-input-label>
    test
  </b2b-input-label>
  <select class="b2b-dropdown__select"></select>
  <span>
    This is a test error
  </span>
</b2b-dropdown>
`;

exports[`should render the dropdown 1`] = `
<b2b-dropdown class="b2b-dropdown">
  <b2b-input-label>
    test
  </b2b-input-label>
  <select class="b2b-dropdown__select"></select>
</b2b-dropdown>
`;

exports[`should render the dropdown with options 1`] = `
<b2b-dropdown class="b2b-dropdown">
  <b2b-input-label>
    test
  </b2b-input-label>
  <select class="b2b-dropdown__select">
    <option>
      An option
    </option>
  </select>
</b2b-dropdown>
`;
